rotkehlchen/tests/api/test_blockchain.py

```diff
@@ -2023,102 +1919,54 @@ def test_add_ksm_blockchain_account_invalid_ens_domain(rotkehlchen_api_server):
 @pytest.mark.parametrize('kusama_manager_connect_at_start', [KUSAMA_TEST_NODES])
 def test_add_ksm_blockchain_account_ens_domain(rotkehlchen_api_server):
     """Test adding a Kusama blockchain account via ENS domain when there is none
-    in the db works as expected, by triggering the logic that attempts to connect
-    to the nodes.
-    """
+    in the db works as expected"""
+
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
     async_query = random.choice([False, True])
-    setup = setup_balances(
-        rotki=rotkehlchen_api_server.rest_api.rotkehlchen,
-        ethereum_accounts=None,
-        btc_accounts=None,
-        eth_balances=None,
-        token_balances=None,
-        btc_balances=None,
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            "blockchainsaccountsresource",
+            blockchain=SupportedBlockchain.KUSAMA.value,
+        ),
+        json={
```

Comment:
I think this test is broken. You are adding an ENS name that should trigger an addition of a kusama account. BUT that ens name should also connect to the kusama node.

You need to:

1. make sure it does without the cheating `@pytest.mark.parametrize('kusama_manager_connect_at_start', [KUSAMA_TEST_NODES])` you have at the start of the test
2. Add a check in the test that after adding the ens address kusama node is connected.

Reply:
I don't think that this test is broken. I checked how substrate manager fixture is created and we try to connect to the nodes at startup only if any kusama accounts are passed (and in this test we don't pass them)

https://github.com/rotki/rotki/blob/develop/rotkehlchen/tests/fixtures/blockchain.py#L212

Reply:
broken it may not be due to the fixture, but you should at least do number 2 of what I asked for.